lmfit/model.py

```diff
@@ -319,17 +319,35 @@ def _get_state(self):
         """Save a Model for serialization.
 
         Note: like the standard-ish '__getstate__' method but not really
-        useful with Pickle.
+        useful with Pickle, and only useful with dill.
+
+        This, and the companion function _buildmodel to use this serialized model
+        now supports versions of 'state'.
+
+        State Versions:
+         '1':  state is a tuple of length 9:
```

Comment:
this is a bit confusing to me as there is actually no version "state 1", at least not explicitly. I assume this state is how things are saved before this PR - correct? I think the description could be a bit more clear on this.

In fact, I am not really sure the additional complication is needed. To me it appears that the "previous" way of doing things would always result in a `tuple`, the "new" method will save things as a `dict`. That distinction in itself should be sufficient to distinguish what to do with the input (and ideally after a while, we can just deprecate and eventually remove reading of the `tuple` state. 
The `version` key seems not really necessary, I would hope that we don't need to change this anymore and we ideally don't want to have to deal with writing/reading of different `versions`. Are you not sure yet about the structure of this yet; it would be good to make sure to have that all settled before implementing this change.

Reply:
@reneeotten Thanks -- will fix